benches/osrank_naive.rs

```diff
@@ -140,17 +141,70 @@ fn bench_osrank_naive_on_small_network(c: &mut Criterion) {
     });
 }
 
-// run with a lower sample size to speed things up
-fn bench_osrank_naive_on_sample_csv(c: &mut Criterion) {
-    let mut network = construct_network(1_000, 10_000);
-    let info = format!("osrank with {:?} nodes, iter: 1", &network.node_count());
-    c.bench(
-        &info,
-        Benchmark::new("sample size 10", move |b| {
-            b.iter(|| run_osrank_naive(&mut network, 1, [0; 16]))
-        })
-        .sample_size(10),
-    );
+fn bench_osrank_naive(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Increasing node count");
+    group.sample_size(10);
+
+    for iter in &[1, 10, 100, 1_000] {
+        for count in &[1_001, 2_501, 5_001, 7_501, 10_001, 15_001] {
+            let mut network = construct_network(*count as usize, 0);
+            let nodes = &network.node_count();
+            group.bench_function(
+                BenchmarkId::from_parameter(nodes),
+                move |b| {
+                    b.iter(|| run_osrank_naive(&mut network, *iter as u32, [0; 16]))
+                }
+            );
+        }
+
+        for count in &[5_000, 10_000, 19_370] {
+            let mut network = construct_network(16_220, *count as usize);
+            let nodes = &network.node_count();
+            group.bench_function(
+                BenchmarkId::from_parameter(nodes),
+                move |b| {
+                    b.iter(|| run_osrank_naive(&mut network, *iter as u32, [0; 16]))
+                }
+            );
+        }
+    }
+    group.finish();
+}
+
+// test iter with managable node count
+fn bench_osrank_naive2(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Increasing iterator");
+    group.sample_size(10);
+
+    for iter in &[1, 10, 100, 1_000] {
+        let mut network = construct_network(1_000, 0);
+        group.bench_function(
+            BenchmarkId::from_parameter(&iter),
+            move |b| {
+                b.iter(|| run_osrank_naive(&mut network, *iter, [0; 16]))
+            }
+        );
+    }
+    group.finish();
+}
+
+fn bench_random_walk(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Random walks with increasing iterator and node count");
+    group.sample_size(10);
+
+    for iter in &[1, 10, 100, 1_000] {
+        for count in &[1_001, 2_501, 5_001, 7_501, 10_001, 15_001] {
+            let network = construct_network(*count as usize, 0);
+            let nodes = &network.node_count();
+            group.bench_function(
+                BenchmarkId::new(format!("Iter {}", &iter), nodes),
+                move |b| {
+                    b.iter(|| run_random_walk(&network, *iter, [0; 16]))
+                }
+            );
+        }
+    }
+    group.finish();
```

Comment:
As we discussed elsewhere, we could try two split the benchmarks in two, by having something like:

```
[[bench]]
name = "osrank_naive_development"
harness = false

[[bench]]
name = "osrank_naive_nightly
harness = false
```

By looking at the structure of `[[bench]]` I think this should be totally fine, as the `[[blah]]` syntax means that cargo expects a list of bench entries. Later, once we have such separation, we could instrument criterion to have the development (fast) ones to be run when we do normal development, and the "slow" for a nightly CI (that I will setup in a separate PR):

```
cargo bench -- dev
cargo bench -- nightly
```

I think it's all it takes!

Reply:
As far as I understand the documentation and from testing, the filter is by `BenchmarkId` so not the file name nor the function name, but the identifier you create for the benchmarks within the functions. This seems a bit repetitive and misplaced info to use for the nightly CI, but I haven't found a different way. To reduce duplication of code, I would keep them in the same file then.